### PR TITLE
CVE-2013-0910 & CVE-2015-1220

### PR DESCRIPTION
**Pull request by lking1106**

### CODE REVIEW

File: cves/CVE-2013-0910.yml [hunk 1]
```diff
@@ -15,11 +15,19 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  This vulnerability was not a found bug, but more of a security oversight. Someone discovered
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: This is good sentence for the beginning of a second paragraph. The "Google Chrome" sentence is a good one to start with.

File: cves/CVE-2013-0910.yml [hunk 2]
```diff
@@ -15,11 +15,19 @@ description_instructions: |
...
+  a potential vulnerability that could be corrected before it was ever an issue.
+  Google Chrome previously did not properly manage browser interactions between browsers and
+  renderers. The issue occurs when plug-ins are loaded or validated/authorized. 
+  Attackers can use it to bypass access restrictions with blocked plug-ins due to
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: I could use a little more detail on this sentence. How does the bypassing actually happen?

File: cves/CVE-2013-0910.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 CVE: CVE-2013-0910
-CWE: 
+CWE: CWE-287
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: This is a pretty broad category - take a look at all the CWE ParentOf relationships. Is one of these more appropriate? Maybe 303?

File: cves/CVE-2013-0910.yml [hunk 4]
```diff
@@ -15,11 +15,19 @@ description_instructions: |
...
+  renderers. The issue occurs when plug-ins are loaded or validated/authorized. 
+  Attackers can use it to bypass access restrictions with blocked plug-ins due to 
+  the exposed process. The renderer is what loads in the plug ins, and is the concerning
+  part here. If a renderer is compromised, there are no longer any restricitions on the
+  type of plug ins that are loaded.
 bounty:
-  date: 
-  amount: 
-  references: []
+  date: Feb 2013
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: Leave the date nil for bounty if none was awarded.

File: cves/CVE-2013-0910.yml [hunk 5]
```diff
@@ -36,17 +44,19 @@ fixes_vcc_instructions: |
...
 - :commit: 8b75315a99bb43615b5efbf56734811bf06cd86b
   :note: ''
-- :commit: 6be31d20e5c3727f136b8dfe06cd7f27ebf898bb
-  :note: ''
-vccs: []
+- :commit: 05f51128a7145375a1661027a9c7f500821c76f5
+  :note: 'This commit is the fix to the revert, actually fixing the bug.'
+vccs: 
+- :commit: ebee9ccaecaf57a0d2b8a36a41de689eef977fea
+  :note: 'This is just the previous commit, because there is not really a direct vulnerablity code introduction. When this vulnerability is introduced, it was the result of a thought experiment on how the code in general is unsafe. Plug-ins were handled this way in many locations, so there would not be one individual commit that caused this.'
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: Huh, ok I'm willing to accept that here - IF, you can give me more details on the history of this feature and this code. Some examples of what you're talking about would be ideal.

File: cves/CVE-2013-0910.yml [hunk 6]
```diff
@@ -141,37 +165,37 @@ lessons:
...
     note: 
   native_wrappers:
-    applies: 
+    applies: false
     note: 
   distrust_input:
-    applies: 
+    applies: false
     note: 
   security_by_obscurity:
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: Can you please add a "sandbox:" entry and add it to this list? I think that's a good tag for us.

File: cves/CVE-2013-0910.yml [hunk 7]
```diff
@@ -85,20 +101,26 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    They were playing with the idea of "breaking out of the Chrome sandbox", and have
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: If this wasn't someone internal to the project, who was it? Is there any evidence of who they are?

File: cves/CVE-2013-0910.yml [hunk 8]
```diff
@@ -85,20 +101,26 @@ discovered:
...
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: |
+    Between the code review and the bug reports, it appeared to all be located in the browser folder 
+    with the code fixes in the plug-in folder.
+  name: 'browser/plugins'
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: Just call this "plugins"

File: cves/CVE-2013-0910.yml [hunk 9]
```diff
@@ -185,4 +209,16 @@ mistakes:
...
+    of code that could be improved, and google responded accordingly. If any mistakes were
+    made, it would be in the design phase. It also might be able to be considered a Maintainability
+    issue, because as technologies evolved, they did not stay upgrade this feature accordingly.
+
+    As far as the mitigations go, the only one mentioned on the cwe page is to use an authentication
+    framework and library. However, this is not entirely relevant because we are not concerned with
+    user authentication, but authentication of a renderer. This is a more unique problem than one that 
+    can be solved with a default library. 
+
+    The fix looks like a pretty good solution for a problem that is not very common.
```
Comment (andymeneely) [inline cves/CVE-2013-0910.yml]: This was a very good report. Thank you!! Keep working at making it better. And hopefully you can provide similar feedback to others in your cohort.

File: cves/CVE-2015-1220.yml [hunk 10]
```diff
@@ -15,7 +15,15 @@ description_instructions: |
...
-description: 
+description: |
+  In Google Chrome, there were multiple occasions where after a memory location is freed,
+  the location is then attempted to be used again. In this specific vulnerability, a 
+  location used for a GIF decoder is attempted to be used again after a gif fails to be
+  decoded. If a remote attacker chooses to attempt to take advantage of this vulnerability, 
+  they could cause a denial of service for other users or other unpredicted behavior. 
+  The way that it is anticipated that an attacker would take advantage of this is by a 
+  "crafted frame size" in a GIF. Any frame size larger than the screen size of a device 
+  could cause the broken GIF to triggers a bug in the decoding code for the GIF.
```
Comment (andymeneely) [inline cves/CVE-2015-1220.yml]: Excellent! This is a solid example of a good description.

File: cves/CVE-2013-0910.yml (hunk 6 above)
Comment (lking1106) [inline cves/CVE-2013-0910.yml]: By this, do you mean add another entry like "native_wrappers" or "security_by_obscurity" already in this list? I just want to make sure I understand what you're asking.

File: cves/CVE-2013-0910.yml (hunk 9 above)
Comment (lking1106) [inline cves/CVE-2013-0910.yml]: Thank you! I appreciate the feedback.

File: cves/CVE-2013-0910.yml (hunk 3 above)
Comment (lking1106) [inline cves/CVE-2013-0910.yml]: I had thought the same thing originally, but were not able to find something that seemed correct. I'll look more into this!

File: cves/CVE-2015-1220.yml [hunk 11]
```diff
@@ -54,9 +64,14 @@ unit_tested:
```
Comment (andymeneely) [inline cves/CVE-2015-1220.yml]: Put 5 upvotes in for me please! This one was interesting to me.

File: cves/CVE-2015-1220.yml [hunk 12]
```diff
@@ -54,9 +64,14 @@ unit_tested:
...
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    This code was unit tested after the vulnerability was found and corrected.
+    In fact, in what I believe to be the vcc, the code writer wrote that it was
+    'simple refactoring, no tests' on the part of the code that they edited. While 
+    they did not introduce the issue, they would have noticed it earlier had they 
+    added tests into the framework at that time.
+  code: false
```
Comment (rjd3756) [inline cves/CVE-2015-1220.yml]: Even before refactoring, there probably should have been tests covering the area that was eventually refactored

File: cves/CVE-2013-0910.yml [hunk 13]
```diff
@@ -15,11 +15,21 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Google Chrome previously did not properly manage browser interactions between browsers and
+  renderers. The issue occurs when plug-ins are loaded or validated/authorized. 
+  Attackers can use it to bypass access restrictions with blocked plug-ins due to 
+  the exposed rendering process. The renderer is what loads the plug ins in the browser, and is
```
Comment (Axiat) [inline cves/CVE-2013-0910.yml]: Small grammatical inconsistency: 
Forget to add a hyphen "-" in the word 'plug in' and 'plug ins' on lines 22,24,25, 85, and 212.

File: cves/CVE-2013-0910.yml [hunk 14]
```diff
@@ -85,20 +103,32 @@ discovered:
...
-  contest: 
+  answer: |
+    Some person with a gmail account, but not a google employee account, was experimenting
+    with the idea of "breaking out of the Chrome sandbox." This goal of this sandbox, according 
+    to the chromium website, is to prevent code from making persistent changes to the computer and 
+    "to provide hard guarantees about what ultimately a piece of code can or cannot do no matter 
+    what its inputs are." 
+    There is no further information about the person that found this vulnerability other than their 
+    gmail account without any personal information. They claim to have experience with Java and 
+    the security(or lack thereof) involved with the language. They found a concern with the process
```
Comment (Axiat) [inline cves/CVE-2013-0910.yml]: No issue with this line, I just wanted to acknowledge this low-key burn +1

File: cves/CVE-2015-1220.yml [hunk 15]
```diff
@@ -54,9 +64,14 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    This code was unit tested after the vulnerability was found and corrected.
+    In fact, in what I believe to be the vcc, the code writer wrote that it was
```
Comment (Axiat) [inline cves/CVE-2015-1220.yml]: I'd suggest changing "the code writer" to "the author"

File: cves/CVE-2013-0910.yml (hunk 13 above)
Comment (lking1106) [inline cves/CVE-2013-0910.yml]: Yeah, it was giving me some troubles in the yaml file. I'll probably remove it here too!

File: cves/CVE-2015-1220.yml (hunk 12 above)
Comment (lking1106) [inline cves/CVE-2015-1220.yml]: Yeah, I agree. They definitely should have added tests when the code was initially written considering this is such a complex system.

**Pull request merged**